### PR TITLE
Update to new qiskit docs theme

### LINKED ISSUE #244: Update docs to use latest qiskit style
E.g. https://github.com/Qiskit-Extensions/circuit-knitting-toolbox/pull/313/files

### PR DESCRIPTION
**Pull request by DanPuzzuoli**
### Summary

Closes #244

Update to new qiskit docs theme.

### CODE REVIEW

**Review by Eric-Arellano — APPROVED** [review]
Thanks!

File: docs/conf.py [hunk 1]
```diff
@@ -54,15 +54,8 @@
 bibtex_footbibliography_header = bibtex_bibliography_header
 
 # html theme options
-html_theme = 'qiskit_sphinx_theme'
-html_last_updated_fmt = '%Y/%m/%d'
```
Comment (Eric-Arellano) [inline docs/conf.py:58]: Oh wait, this should still be there.

Comment (DanPuzzuoli) [reply docs/conf.py:58]: You mean the `html_last_updated_fmt` should remain?

Comment (DanPuzzuoli) [reply docs/conf.py:58]: (the updated `html_theme` is being displayed at the bottom of this change block)

File: requirements-dev.txt [hunk 2]
```diff
@@ -2,7 +2,7 @@ stestr>=3.0.0
 astroid==2.9.3
 pylint==2.12.2
 black~=22.0
-qiskit-sphinx-theme~=1.12.0
+qiskit-sphinx-theme~=1.13.0
```
Comment (Eric-Arellano) [inline requirements-dev.txt:5]: Let's you use the ecosystem theme:

```suggestion
qiskit-sphinx-theme~=1.14.0rc1
```

File: docs/conf.py [hunk 3]
```diff
@@ -54,15 +54,9 @@
 bibtex_footbibliography_header = bibtex_bibliography_header
 
 # html theme options
-html_theme = 'qiskit_sphinx_theme'
+html_theme = 'qiskit'
```
Comment (Eric-Arellano) [inline docs/conf.py:57]:
```suggestion
html_theme = 'qiskit-ecosystem'
```

**Comment by arnaucasau** [comment]
@DanPuzzuoli @Eric-Arellano The API has been reorganised in https://github.com/Qiskit-Extensions/qiskit-dynamics/pull/264. After updating this branch with the new changes on the main branch, we should be able to see if we have achieved the improvement needed 👍

File: requirements-dev.txt [hunk 4]
```diff
@@ -2,7 +2,7 @@ stestr>=3.0.0
 astroid==2.9.3
 pylint==2.12.2
 black~=22.0
-qiskit-sphinx-theme~=1.12.0
+qiskit-sphinx-theme~=1.14.0rc1
```
Comment (Eric-Arellano) [inline requirements-dev.txt:5]:
```suggestion
qiskit-sphinx-theme~=1.16.0
```

**Comment by DanPuzzuoli** [comment]
@Eric-Arellano is this good to go? Thanks for both of your help @Eric-Arellano and @arnaucasau

**Comment by Eric-Arellano** [comment]
@arnaucasau wanted to check on Monday that the build times and page sizes are acceptable. He's OOO now (Spain hours), but DMed me that he plans to review on Monday :)

**Review by arnaucasau — APPROVED** [review]
LGTM!

**Comment by Eric-Arellano** [comment]
@DanPuzzuoli fyi neither of us have write access. Not sure if you need someone else to approve this also.

**Comment by DanPuzzuoli** [comment]
Thanks @arnaucasau @Eric-Arellano . You both know best for this stuff, so with your approval I'll just bypass branch protections for this one.

**Pull request merged**